integration-tests/src/test/java/com/datastax/oss/driver/api/core/cql/ExecutionInfoWarningsIT.java

```diff
@@ -169,24 +165,36 @@ public void should_expose_warnings_on_execution_info() {
                 + "APPLY BATCH",
             Strings.repeat("1", 2 * 1024), Strings.repeat("1", 3 * 1024));
     Statement<?> st = SimpleStatement.builder(String.format(query)).build();
-    ResultSet result = SESSION_RULE.session().execute(st);
+    ResultSet result = sessionRule.session().execute(st);
     ExecutionInfo executionInfo = result.getExecutionInfo();
     assertThat(executionInfo).isNotNull();
     List<String> warnings = executionInfo.getWarnings();
     assertThat(warnings).isNotEmpty();
     // verify the log was generated
-    verify(appender, after(500).times(1)).doAppend(loggingEventCaptor.capture());
+    verify(appender, after(500).atLeast(1)).doAppend(loggingEventCaptor.capture());
     assertThat(loggingEventCaptor.getValue().getMessage()).isNotNull();
-    String logMessage = loggingEventCaptor.getValue().getFormattedMessage();
-    assertThat(logMessage)
-        .startsWith("Query '")
-        // query will only be logged up to MAX_QUERY_LENGTH
-        // characters
-        .contains(query.substring(0, RequestLogger.DEFAULT_REQUEST_LOGGER_MAX_QUERY_LENGTH))
-        .contains("' generated server side warning(s): ")
-        .contains(
-            String.format(
-                "Batch for [%s.test] is of size 5152, exceeding specified threshold of 5120 by 32.",
-                SESSION_RULE.keyspace().asCql(true)));
+    boolean testPassed = false;
+    for (ILoggingEvent log : loggingEventCaptor.getAllValues()) {
+      // some versions of Cassandra return multiple warnings for this query. We're only verifying
+      // the common pieces of the warnings.
+      String logMessage = log.getFormattedMessage();
+      // all logs should start with the driver log prefix
+      assertThat(logMessage)
+          .startsWith("Query '")
+          // query will only be logged up to MAX_QUERY_LENGTH characters
+          .contains(query.substring(0, RequestLogger.DEFAULT_REQUEST_LOGGER_MAX_QUERY_LENGTH))
+          .contains("' generated server side warning(s): ");
+      // we're now only looking for:
+      // "Batch ... for [KS.test] is of size .... exceeding specified threshold"
+      if (logMessage.contains("Batch")) {
```

Comment:
Why not create an assertion for that? E.g.:

```
assertThat(logMessage)
    .contains("Batch")
    .contains(String.format("for [%s.test] is of size", sessionRule.keyspace().asCql(true)))
    .contains("exceeding specified threshold");
```
Besides, you are affecting true to `testPassed` inside a loop, which means that as long as the variable is set to true once, the whole test will pass, which imo is not what we want. Iow I think it would be better to get rid of the `testPassed`  variable.

Reply:
The reason for the loop and the flag is related to the multiple logs issue for certain versions of Cassandra. For versions 2.2 and 3.0, when 2 logs are generated, one of the logs does not contain any of the above assertions. And I'm not sure I am guaranteed the same order in the log capture. So I loop through the logs here, and when I come across one that has "Batch", I assert the rest of the message, set a flag and break out of the loop on line 195 below.

In the case where it's an older version of Cassandra, and I get two logs, this skips over the other log that doesn't appear in 3.11, either because it it doesn't contain "Batch", or the loop was broken out of on the log that did.

Reply:
I see. Would the following work?

```
List<String> logMessages =
    loggingEventCaptor.getAllValues().stream()
        .map(ILoggingEvent::getFormattedMessage)
        .collect(Collectors.toList());
assertThat(logMessages)
    .hasOnlyOneElementSatisfying(
        logMessage -> {
          assertThat(logMessage)
              .contains("Batch")
              .contains(
                  String.format("for [%s.test] is of size", sessionRule.keyspace().asCql(true)))
              .contains("exceeding specified threshold");
        });
```

Reply:
I can change it up to do this.

Reply:
So I tried implementing this, but it still fails as `hasOnlyOneElementSatisfying` will fail the assertion if the List of logMessages has more than one element. I thought that method would verify that the list only contained one element that passes the assertion, but it in fact asserts that there is only one element AND that it passes the provided assertion.
I'm not seeing an easy way to assert that, no matter how many logs are in the Captor, there is only one that passes this specific set of assertions, without looping through the logs.